bridge/client/app/_components/ktb-edit-service/ktb-edit-service.component.ts

```diff
@@ -40,6 +46,73 @@ export class KtbEditServiceComponent implements OnDestroy {
       this.eventService.deletionProgressEvent.next({isInProgress: true});
       this.deleteService();
     });
+
+    this.project$?.pipe(
+      takeUntil(this.unsubscribe$),
+    ).subscribe(project => {
+      if (!this.fileTree && project && (project.gitRemoteURI || project.gitRemoteURI !== '') && this.serviceName) {
+        this.getResourcesAndTransform(project, this.serviceName);
+      }
+    });
+  }
+
+  public getResourcesAndTransform(project: Project, serviceName: string): void {
+    this.dataService.getServiceResourceForAllStages(project.projectName, serviceName).subscribe((resources) => {
+      this.fileTree = [];
+      project.stages.forEach(stage => {
+        this.fileTree?.push({stage: stage.stageName, files: this.getFileTreeForStage(resources, stage.stageName)});
```

Comment:
You could change it to a simple `for-loop` (`for const stage of project.stages`). Then it is not necessary to add optional chaining here.
Another option would be `this.fileTree = project.stages.map(stage => ...)`